helper.py

```diff
@@ -1,41 +1,186 @@
-#!/usr/bin/env python3
+"""
+IMPORTANT NOTE!!
 
-# This file may be obsolete - need to check if scaling in shephard is laggy
+=> The following will ONLY run on Python 3.7 to 3.9
+=> Install websockets version 11.0.3 via pip whl (do not use v12)
+=> Install asynio 0.2.0 via pip whl
+"""
 
-import os
+import asyncio
+import aionotify
+from websockets.server import serve
+import websockets
 import time
-
+import base64
+import os
+from io import BytesIO
 from PIL import Image
 
 
-static_path = 'shepherd/static/'
-input_file = static_path + "full_image.jpg"
-output_file = static_path + "image.jpg"
-stat_file = static_path + "imgtime.txt"
+CONNECTIONS = set()
+
+# The following sets up the asynchronous waiting for file change
+picture_watcher = aionotify.Watcher()
+log_watcher = aionotify.Watcher()
+
+img_static_path = "/home/pi/shepherd/shepherd/static/"
+img_input_file = img_static_path + "image.jpg"
+img_output_file = img_static_path + "resized_image.jpg"
+stat_file = img_static_path + "imgtime.txt"
 mod_time = 0
 out_width = 640.0
 out_height = 480.0
 
-while True:
-    time.sleep(0.1)
-    if os.path.exists(input_file):
-        stat_obj = os.stat(input_file)
-        if mod_time != stat_obj.st_mtime:
-            try:
-                img = Image.open(input_file)
-                width_i, height_i = img.size
-                width = width_i * 1.0
-                height = height_i * 1.0
-                scale_x = out_width / width
-                scale_y = out_height / height
-                if scale_x > scale_y:
-                    img = img.resize((int(width*scale_y), int(height*scale_y)), Image.ANTIALIAS)
-                else:
-                    img = img.resize((int(width*scale_x), int(height*scale_x)), Image.ANTIALIAS)
-
-                mod_time = stat_obj.st_mtime
-                img.save(output_file,"JPEG")
-                with open(stat_file, "w") as myfile:
-                    myfile.write(str(mod_time))
-            except IOError:
-                pass
+log_static_path = "/media/RobotUSB/"
+log_input_file = log_static_path + "logs.txt"
+
+picture_watcher.watch(
+    alias="image", path=img_input_file, flags=aionotify.Flags.MODIFY
+)  # sets up watcher
+log_watcher.watch(alias="logs", path=log_input_file, flags=aionotify.Flags.MODIFY)
+
+
+def shrink_image(img):
+    width_i, height_i = img.size
+    width = float(width_i)
+    height = float(height_i)
+    scale_x = out_width / width
+    scale_y = out_height / height
+    if scale_x > scale_y:
+        img = img.resize((int(width * scale_y), int(height * scale_y)), Image.LANCZOS)
+    else:
+        img = img.resize((int(width * scale_x), int(height * scale_x)), Image.LANCZOS)
+    return img
```

Comment:
This works and is perfectly fine. To avoid the if statement/repeated code in the future you could consider something like:

```py
scale_factor = min(scale_x, scale_y)
return img.resize((int(width * scale_factor), int(height * scale_factor)), Image.LANCZOS)
```

Reply:
Also a comment as to what LANCZOS is might be useful here.